sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/IdempotentPublishingLiveTests.cs

```diff
@@ -0,0 +1,546 @@
+                cancellationSource.CancelAfter(EventHubsTestEnvironment.Instance.TestExecutionTimeLimit);
+
+                async Task sendEvents(string partition, int delayMilliseconds)
+                {
+                    await Task.Delay(delayMilliseconds);
+                    await producer.SendAsync(EventGenerator.CreateEvents(5), new SendEventOptions { PartitionId = partition }, cancellationSource.Token);
+                }
+
+                var partitions = await producer.GetPartitionIdsAsync(cancellationSource.Token);
+                var pendingSends = new List<Task>();
+
+                foreach (var partition in partitions)
+                {
+                    pendingSends.Add(sendEvents(partition, 50));
+                    pendingSends.Add(sendEvents(partition, 0));
+                }
+
+                Assert.That(async () => await Task.WhenAll(pendingSends), Throws.Nothing);
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducerClient" /> is able to
+        ///   perform operations when the idempotent publishing feature is enabled.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerSequencesEvents()
+        {
+            await using (EventHubScope scope = await EventHubScope.CreateAsync(2))
+            {
+                var connectionString = EventHubsTestEnvironment.Instance.BuildConnectionStringForEventHub(scope.EventHubName);
+                var options = new EventHubProducerClientOptions { EnableIdempotentPartitions = true };
+
+                await using var producer = new EventHubProducerClient(connectionString, options);
+
+                var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(EventHubsTestEnvironment.Instance.TestExecutionTimeLimit);
+
+                var partition = (await producer.GetPartitionIdsAsync(cancellationSource.Token)).Last();
+                var sendOptions = new SendEventOptions { PartitionId = partition };
+
+                var partitionProperties = await producer.GetPartitionPublishingPropertiesAsync(partition);
+                var eventSequenceNumber = partitionProperties.LastPublishedSequenceNumber;
+
+                var events = EventGenerator.CreateEvents(10).ToArray();
+                Assert.That(events.Any(item => item.PublishedSequenceNumber.HasValue), Is.False, "Events should start out as unpublished with no sequence number.");
+
+                await producer.SendAsync(events, sendOptions, cancellationSource.Token);
+                Assert.That(events.All(item => item.PublishedSequenceNumber.HasValue), Is.True, "Events should be sequenced after publishing.");
+
+                foreach (var item in events)
+                {
+                    Assert.That(item.PublishedSequenceNumber, Is.EqualTo(++eventSequenceNumber), $"The sequence numbers should be contiguous.  Event { eventSequenceNumber + 1 } was out of order.");
```

Comment:
nit: I'm not sure I understand why the assert message specifies that `eventSequenceNumber + 1` is the one out of order. Wouldn't it be the current `eventSequenceNumber` as incremented in the `EqualTo` statement?

Reply:
d'oh.   Yup, that's a mistake.